src/ControllerV2/BeneficiaryPersonalDataController.php

```diff
@@ -0,0 +1,452 @@
+            'beneficiary' => $beneficiary,
+            'event' => $event,
+        ]);
+    }
+
+    #[Route(path: '/{id}/documents', name: 'document_list', requirements: ['id' => '\d+'], methods: ['GET'])]
+    #[IsGranted('UPDATE', 'beneficiary')]
+    public function listDocuments(
+        Request $request,
+        Beneficiaire $beneficiary,
+        FolderableItemManager $manager,
+        PaginatorService $paginator,
+    ): Response {
+        return $this->renderForm('v2/vault/document/index.html.twig', [
+            'beneficiary' => $beneficiary,
+            'foldersAndDocuments' => $paginator->create(
+                $manager->getFoldersAndDocumentsWithUrl($beneficiary),
+                $request->query->getInt('page', 1),
+            ),
+            'form' => $this->getSearchForm(
+                $this->generateUrl('document_search', ['id' => $beneficiary->getId()]),
+            )->handleRequest($request),
+        ]);
+    }
+
+    #[Route(
+        path: '/{id}/documents/upload',
+        name: 'document_upload',
+        requirements: ['id' => '\d+'],
+        methods: ['POST'],
+    )]
+    #[IsGranted('UPDATE', 'beneficiary')]
+    public function upload(
+        Request $request,
+        Beneficiaire $beneficiary,
+        DocumentManager $manager,
+        DossierRepository $folderRepository
+    ): Response {
+        if (!$files = $request->files->get('files')) {
+            return new Response(null, 400);
+        }
+        $folderId = $request->query->get('folder');
+
+        $manager->uploadFiles(
+            $files,
+            $beneficiary,
+            $folderId ? $folderRepository->find($folderId) : null
+        );
+
+        return new Response(null, 201);
+    }
+
+    #[Route(
+        path: '/{id}/documents/search',
+        name: 'document_search',
+        requirements: ['id' => '\d+'],
+        methods: ['POST'],
+        condition: 'request.isXmlHttpRequest()',
+    )]
+    #[IsGranted('UPDATE', 'beneficiary')]
+    public function searchDocuments(
+        Request $request,
+        Beneficiaire $beneficiary,
+        FolderableItemManager $manager,
+        PaginatorService $paginator,
+    ): JsonResponse {
+        $searchForm = $this->getSearchForm(
+            $this->generateUrl('document_search', ['id' => $beneficiary->getId()]),
+        )->handleRequest($request);
+
+        $search = $searchForm->get('search')->getData();
+
+        return new JsonResponse([
+            'html' => $this->renderForm('v2/vault/document/_list.html.twig', [
+                'foldersAndDocuments' => $paginator->create(
+                    $manager->getFoldersAndDocumentsWithUrl($beneficiary, null, $search),
+                    $request->query->getInt('page', 1),
+                ),
+                'beneficiary' => $beneficiary,
+                'form' => $searchForm,
+            ])->getContent(),
+        ]);
+    }
+
+    #[Route(
+        path: '/{id}/folder/{folderId}/search',
```

Comment:
You should name it ${parendFolderId}, And maybee, for more consistency with all orther routes, it could be a post param instead of a path param